src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -373,11 +536,7 @@ class RaftIvfIndexNode : public IndexNode {
 
     virtual int64_t
     Count() const override {
-        auto result = std::int64_t{};
-        if (gpu_index_) {
-            result = gpu_index_->size();
-        }
-        return result;
+        return counts_;
```

Comment:
Is `Count` not supposed to return the total number of rows in the index? It looks like `counts_` is getting overridden with each `Add` call.

Reply:
hello,  when we plug knowhere into milvus, milvus will check the count of the original dataset, I knowhere raft ivf will do some data padding, so, when we get the size from the raft index, milvus will crash down.

Reply:
Could you give us a little more detail on this? `index.size()` should return the actual size of the index without padding. Are you seeing a specific mismatch between the reported size and the number of added vectors? If you have reproducer code, that would be even better.

Reply:
My apologies! I just realized that what I said is true in 23.04 but false in 23.02 (which we are currently using. I'd recommend we upgrade for this and the other 23.04 features we've touched on elsewhere. I'm trying to get this PR to build right now, but once I've done so, I'll try to contribute code suggestions for how best to make use of 23.04.